frontend/src/components/membership/GroupMembership.js

```diff
@@ -0,0 +1,371 @@
+              )}
+            </For>
+          </thead>
+          <tbody>
+            <For each={table.getRowModel().rows}>
+              {row => {
+                return (
+                  <tr class="border-b bg-white dark:border-gray-700 dark:bg-gray-800">
+                    <For each={row.getVisibleCells()}>
+                      {cell => {
+                        return (
+                          <td class="px-6 py-4">
+                            {flexRender(
+                              cell.column.columnDef.cell,
+                              cell.getContext()
+                            )}
+                          </td>
+                        );
+                      }}
+                    </For>
+                  </tr>
+                );
+              }}
+            </For>
+          </tbody>
+        </table>
+        <div class="mt-4 flex items-center justify-center gap-2">
+          <button
+            onClick={() => table.previousPage()}
+            disabled={!table.getCanPreviousPage()}
+            class="flex h-8 w-24 items-center justify-center rounded-lg border border-gray-300 bg-white px-3 text-sm font-medium text-gray-500 hover:bg-gray-100 hover:text-gray-700 disabled:text-gray-300 dark:border-gray-700 dark:bg-gray-800 dark:text-gray-400 dark:hover:bg-gray-700 dark:hover:text-white"
+          >
+            <ChevronLeft width={20} />
+            Previous
+          </button>
+          <span class="flex items-center gap-1">
+            <div>Page</div>
+            <strong>
+              {table.getState().pagination.pageIndex + 1} of{" "}
+              {table.getPageCount().toLocaleString()}
+            </strong>
+          </span>
+          <button
+            onClick={() => table.nextPage()}
+            disabled={!table.getCanNextPage()}
+            class="flex h-8 w-24 items-center justify-center rounded-lg border border-gray-300 bg-white px-3 text-sm font-medium text-gray-500 hover:bg-gray-100 hover:text-gray-700 disabled:text-gray-300 dark:border-gray-700 dark:bg-gray-800 dark:text-gray-400 dark:hover:bg-gray-700 dark:hover:text-white"
+          >
+            Next
+            <ChevronRight width={20} />
+          </button>
+        </div>
+      </Show>
+      <p class="my-2">{status()}</p>
+    </>
+  );
+};
+
+const GroupMembership = props => {
+  const [status, setStatus] = createSignal();
+
+  const [payingPlayers, setPayingPlayers] = createSignal([]);
+  const [paymentSuccess, setPaymentSuccess] = createSignal(false);
+
+  const paymentSuccessCallback = () => {
+    setPaymentSuccess(true);
+  };
+
+  const handlePlayerPayingStatus = (player, isPaying) => {
+    // Don't perform any actions on a player who already has a membership
+    if (player?.has_membership) {
+      return;
+    }
+    if (isPaying) {
+      // Add player to paying Players
+      setPayingPlayers([
+        ...payingPlayers().filter(p => p.id !== player.id),
+        player
+      ]);
+    } else {
```

Comment:
do we need to the filter here ?

Reply:
yeah just so we dont duplicate by mistake also

Reply:
hmm ok